test/e2e/md_scale.go

```diff
@@ -123,6 +124,17 @@ func MachineDeploymentScaleSpec(ctx context.Context, inputGetter func() MachineD
 			Replicas:                  1,
 			WaitForMachineDeployments: input.E2EConfig.GetIntervals(specName, "wait-worker-nodes"),
 		})
+
+		By("Deleting the MachineDeployment with foreground deletion")
```

Comment:
Just added this test to verify that foreground deletion would work at the moment.

Long-term we would like to have the same behavior as with the Cluster (that MD is deleted "in the foreground" independent of if foreground or background deletion is used)

Reply:
Do we have an issue to discuss "Implement forced foreground deletion"?

Reply:
I would create one.

Reply:
xref: https://github.com/kubernetes-sigs/cluster-api/issues/10710